libevmasm/Assembly.cpp

```diff
@@ -1156,6 +1283,130 @@ LinkerObject const& Assembly::assemble() const
 	return ret;
 }
 
+std::map<uint8_t, uint8_t> Assembly::createSubIdsReplacements() const
+{
+	std::set<uint8_t> referencedSubcontainersIds;
+	solAssert(m_subs.size() < std::numeric_limits<uint8_t>::max());
```

Comment:
I see in the spec that `num_container_sections` is 16 bit and ranges from 1 to 256.

So shouldn't 255 still be allowed here?

```suggestion
	solAssert(m_subs.size() <= std::numeric_limits<uint8_t>::max());
```

And shouldn't we rather use `uint16_t` for IDs and add 1? Or is that meant to happen somewhere else?

Reply:
BTW, I didn't really expect you to include even this stub. Since removal of unreferenced containers is optional, I'd be fine with this only being added later with full implementation. I mean, I don't mind having it already here either, but just saying.

Reply:
Shit. U r right. I though there EOF spec reserve only one byte for num of container sections. I need to change it to 2 bytes then. It can be in the range `1 - 256`

Reply:
Also, would be nice if the name actually indicated that the purpose is the removal of unreferenced subs. How about something like `findReferencedContainers()` or `remapReferencedContainers()`?

Reply:
`findReferencedContainers()` used it at the beginning but changed after :) Now it's back as initially.